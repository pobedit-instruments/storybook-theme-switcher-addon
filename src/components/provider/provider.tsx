import React, {ReactNode, useState} from 'react';
import {getTheme} from '../../utils/storage';
import {listenTheme} from '../../hooks';
import {Theme} from '../../config';

type Props = {
    children: ReactNode;
};

export const ThemeContext = React.createContext(Theme.LIGHT.toString());

export const ThemeProvider: React.FunctionComponent<Props> = (props) => {
    const [theme, setTheme] = useState<string>(getTheme());

    listenTheme(() => {
        setTheme(getTheme());
    });

    return <ThemeContext.Provider {...props} value={theme} />;
};
