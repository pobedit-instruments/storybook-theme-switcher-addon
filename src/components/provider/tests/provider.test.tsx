import React from 'react';
import {mount} from 'enzyme';

import {ThemeProvider} from '../provider';

describe('Test Button component', () => {
    it('renders', () => {
        const wrapper = mount(<ThemeProvider>Button</ThemeProvider>);

        expect(wrapper).toMatchSnapshot();
    });

    // it('user click on button', () => {
    //     const eventHandler = jest.fn();
    //     const wrapper = mount(<Button onClick={eventHandler} />);
    //
    //     wrapper.simulate('click');
    //
    //     expect(eventHandler).toHaveBeenCalled();
    // });
});
