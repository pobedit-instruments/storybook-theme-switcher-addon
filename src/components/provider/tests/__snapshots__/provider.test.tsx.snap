// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button component renders 1`] = `
<ThemeProvider>
  Button
</ThemeProvider>
`;
